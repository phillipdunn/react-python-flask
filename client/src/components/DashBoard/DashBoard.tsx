import { Box, Button, Skeleton, Typography } from '@mui/material';
import React, { FC, useEffect, useState } from 'react';

import { BarChart } from '@mui/x-charts';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { copyToClipboard, dataCleaner } from '../../helpers';
import EnhancedTable from '../Table/Table';
export interface Data {
  message_id: number;
  timestamp: string;
  credits_used: number;
  report_name?: string;
}

interface Props { }

const DashBoard: FC<Props> = () => {
  const [data, setData] = useState<Data[]>([]);
  useEffect(() => {
    fetch('/usage')
      .then((res) => res.json())
      .then((data) => {
        setData(data.usage)
      });
  }, []);

  const notify = () => toast("Link to dashboard copied to clipboard");
  return (
    < Box sx={{ m: 3 }}>
      <Box sx={{ display: 'flex', justifyContent: 'space-between' }}>
        <Typography variant="h4" sx={{ pb: 3, }}>Credit Usage Dashboard
        </Typography>
        <Box>
          <Button variant='outlined' onClick={() => copyToClipboard(window.location.href, notify)}>Share Dashboard</Button></Box>
      </Box>
      <ToastContainer />
      {data.length === 0 ?
        <>
          <Skeleton variant="rectangular" width={1800} height={500} />
          <Skeleton variant="rectangular" width={1800} height={400} sx={{ mt: 4 }} />
        </> : <>
          <Box sx={{ py: 3 }}>
            <BarChart
              xAxis={[{ scaleType: 'band', dataKey: 'timestamp' }]}
              series={[{ label: 'Credit usage', dataKey: 'creditsUsed' }]}
              dataset={dataCleaner(data)}
              width={1800}
              height={500}
              grid={{ horizontal: true }}
              borderRadius={8} />
          </Box>
          <EnhancedTable rows={data} /></>
      }
    </Box>
  );
};

export default DashBoard;
