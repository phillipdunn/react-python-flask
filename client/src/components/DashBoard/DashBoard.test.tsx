import React from 'react';
import { faker } from '@faker-js/faker';
import { act, create, ReactTestRenderer } from 'react-test-renderer';
import DashBoard from './DashBoard';

faker.seed(1);

describe('<DashBoard/> component', () => {
test('renders component', () => {
let renderer: ReactTestRenderer;

act(() => {
renderer = create(<DashBoard />);
});

expect(renderer!.toJSON()).toMatchSnapshot('Initial');
});
});
