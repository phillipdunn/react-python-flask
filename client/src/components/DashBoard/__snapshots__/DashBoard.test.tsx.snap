// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashBoard/> component renders component: Initial 1`] = `
<div
  className="MuiBox-root css-yuob64"
>
  <div
    className="MuiBox-root css-gg4vpm"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 css-tjvwm3-MuiTypography-root"
    >
      Credit Usage Dashboard
    </h4>
    <div
      className="MuiBox-root css-0"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Share Dashboard
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    className="Toastify"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
  <span
    className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
    style={
      Object {
        "height": 500,
        "width": 1800,
      }
    }
  />
  <span
    className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-gepztz-MuiSkeleton-root"
    style={
      Object {
        "height": 400,
        "width": 1800,
      }
    }
  />
</div>
`;
